build/lib/pdbvangogh/models.py

```diff
@@ -0,0 +1,136 @@
+    style_loss *= style_weight / num_style_layers
+
+    content_loss = tf.add_n([tf.reduce_mean((content_outputs[name] - content_targets[name]) ** 2) for name in content_outputs.keys()])
+    content_loss *= content_weight / num_content_layers
+    loss = style_loss + content_loss
+    return loss
+
+
+def am_i_in_a_jupyter_notebook():
+    try:
+        from IPython import get_ipython
+
+        # Check if the get_ipython function exists
+        # If we're in Jupyter, get_ipython will not be None
+        if "IPKernelApp" not in get_ipython().config:  # pragma: no cover
+            return False
+    except ImportError:
+        # IPython is not installed, definitely not in a notebook
+        return False
+    except AttributeError:
+        # IPython is installed but we're not in a Jupyter environment
+        return False
+    return True
+
+
+def style_transfer_vgg19(content_image, style_image, hyperparameters=data_models.vgg19_transfer_parameters()):
+    x = tf.keras.applications.vgg19.preprocess_input(content_image * 255)
+    x = tf.image.resize(x, (224, 224))
+    vgg = tf.keras.applications.VGG19(include_top=False, weights="imagenet")
+    content_layers = ["block5_conv2"]
+    style_layers = ["block1_conv1", "block2_conv1", "block3_conv1", "block4_conv1", "block5_conv1"]
+    num_content_layers = len(content_layers)
+    num_style_layers = len(style_layers)
+    style_extractor = vgg_layers(style_layers)
+    style_outputs = style_extractor(style_image * 255)
+    extractor = StyleContentModel(style_layers, content_layers)
+    results = extractor(tf.constant(content_image))
+    style_targets = extractor(style_image)["style"]
+    content_targets = extractor(content_image)["content"]
+    start = time.time()
+    image = tf.Variable(content_image)
+
+    @tf.function()
+    def train_step(image, style_targets, content_targets):
+        with tf.GradientTape() as tape:
+            outputs = extractor(image)
+            loss = style_content_loss(outputs, style_targets, content_targets, num_content_layers, num_style_layers, hyperparameters.style_weight, hyperparameters.content_weight)
+            if hyperparameters.total_variation_weight > 0:
+                loss += hyperparameters.total_variation_weight * tf.image.total_variation(image)
+
+        grad = tape.gradient(loss, image)
+        opt.apply_gradients([(grad, image)])
+        image.assign(clip_0_1(image))
+
+    epochs = hyperparameters.epochs
+    steps_per_epoch = hyperparameters.steps_per_epoch
+    opt = tf.keras.optimizers.Adam(learning_rate=0.02, beta_1=0.99, epsilon=1e-1)
+    step = 0
+    in_jupyter_notebook = am_i_in_a_jupyter_notebook()
+    for n in range(epochs):
+        for m in range(steps_per_epoch):
+            step += 1
+            train_step(image, style_targets, content_targets)
+            print(".", end="", flush=True)
+        # if in_jupyter_notebook:
```

Comment:
nit: clean this up or uncomment?  Since you already go through the pains of knowing whether you are in an interactive context I'm guessing should be safe?